_articles/aws_sts_storage_locations.md

```diff
@@ -0,0 +1,178 @@
+
+{% include note.html content="You can only create STS Storage Locations on an empty folder. This is to ensure consistency between the STS Storage Location and the Synapse folder." %}
+
+## Synapse-Managed STS Storage Locations
+
+You can create an STS Storage Location using Synapse storage. Synapse will store the files for you, but you can still get temporary S3 credentials, scoped to just the files and folders within your STS Storage Location.
+
+To set up the STS Storage Location using Synapse storage, first make sure you have an empty Synapse folder, then run the following code:
+
+##### Python
+
+```python
+# Set storage location
+import synapseclient
+import json
+syn = synapseclient.login()
+FOLDER = 'syn12345'
+
+destination = {'uploadType':'S3',
+               'stsEnabled':True,
+               'concreteType':'org.sagebionetworks.repo.model.project.S3StorageLocationSetting'}
+destination = syn.restPOST('/storageLocation', body=json.dumps(destination))
+
+project_destination ={'concreteType': 'org.sagebionetworks.repo.model.project.UploadDestinationListSetting',
+                      'settingsType': 'upload'}
+project_destination['locations'] = [destination['storageLocationId']]
+project_destination['projectId'] = FOLDER
+
+project_destination = syn.restPOST('/projectSettings', body = json.dumps(project_destination))
+```
+
+##### R
+
+```r
+#set storage location
+library(synapser)
+library(rjson)
+synLogin()
+folderId <- 'syn12345'
+
+destination <- list(uploadType='S3',
+                    stsEnabled=TRUE,
+                    concreteType='org.sagebionetworks.repo.model.project.S3StorageLocationSetting')
+destination <- synRestPOST('/storageLocation', body=toJSON(destination))
+
+projectDestination <- list(concreteType='org.sagebionetworks.repo.model.project.UploadDestinationListSetting',
+                           settingsType='upload')
+projectDestination$locations <- list(destination$storageLocationId)
+projectDestination$projectId <- folderId
+
+projectDestination <- synRestPOST('/projectSettings', body=toJSON(projectDestination))
+```
+
+You can then upload files to this STS Storage Location using the normal methods, either through [the Synapse website](https://www.synapse.org/) or through the client of your choice.
+
+## Custom STS Storage Locations
+
+You can also create an STS Storage Location using a custom storage location in an external AWS S3 bucket. To set this up, first follow the steps in [Custom Storage Locations](custom_storage_location.md), up until the step "Set S3 Bucket as Upload Location". Instead, make sure you have an empty Synapse folder, then run the following code.
+
+{% include important.html content="You should specify a baseKey in your storage location, which is a folder path in your bucket that all files in the storage location should go into. The baseKey is optional, but if it is not specified, the temporary AWS credentials vended by STS will give users access to the whole bucket." %}
```

Comment:
```suggestion
{% include important.html content="If a baseKey is **not** specified, the temporary AWS credentials vended by STS will give users access to the whole bucket. To prevent access to the whole bucket, enter a folder path in your bucket that all files in the storage location should go into as the baseKey. " %}
```